akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala

```diff
@@ -4,104 +4,116 @@
 
-        persistentActor ! cmd
+      val persistenceId = PersistenceId.ofUniqueId("your-persistence-id")
+      val persistentActor = spawn(
+        EventSourcedBehavior[Cmd, Evt, State](
+          persistenceId,
+          emptyState = State.empty,
+          commandHandler = (_, cmd) => Effect.persist(Evt(cmd.data)),
+          eventHandler = (state, evt) => state.updated(evt)))
+      val cmd = Cmd("data")
 
-        val expectedPersistedEvent = Evt(cmd.data)
-        persistenceTestKit.expectNextPersisted("your-persistence-id", expectedPersistedEvent)
-      }
+      persistentActor ! cmd
 
+      val expectedPersistedEvent = Evt(cmd.data)
+      persistenceTestKit.expectNextPersisted(persistenceId.id, expectedPersistedEvent)
     }
-  }
-  //#testkit-typed-usecase
-
-  //#set-event-storage-policy
-  class SampleEventStoragePolicy extends EventStorage.JournalPolicies.PolicyType {
-
-    //you can use internal state, it does not need to be thread safe
-    var count = 1
-
-    override def tryProcess(persistenceId: String, processingUnit: JournalOperation): ProcessingResult =
-      if (count < 10) {
-        count += 1
-        //check the type of operation and react with success or with reject or with failure.
-        //if you return ProcessingSuccess the operation will be performed, otherwise not.
-        processingUnit match {
-          case ReadEvents(batch) if batch.nonEmpty => ProcessingSuccess
-          case WriteEvents(batch) if batch.size > 1 =>
-            ProcessingSuccess
-          case ReadSeqNum      => StorageFailure()
-          case DeleteEvents(_) => Reject()
-          case _               => StorageFailure()
-        }
-      } else {
-        ProcessingSuccess
-      }
 
   }
-  //#set-event-storage-policy
-
-  //#set-snapshot-storage-policy
-  class SampleSnapshotStoragePolicy extends SnapshotStorage.SnapshotPolicies.PolicyType {
-
-    //you can use internal state, it does not need to be thread safe
-    var count = 1
-
-    override def tryProcess(persistenceId: String, processingUnit: SnapshotOperation): ProcessingResult =
-      if (count < 10) {
-        count += 1
-        //check the type of operation and react with success or with reject or with failure.
-        //if you return ProcessingSuccess the operation will be performed, otherwise not.
-        processingUnit match {
-          case ReadSnapshot(_, payload) if payload.nonEmpty =>
-            ProcessingSuccess
-          case WriteSnapshot(meta, payload) if meta.sequenceNr > 10 =>
-            ProcessingSuccess
-          case DeleteSnapshotsByCriteria(_) => StorageFailure()
-          case DeleteSnapshotByMeta(meta) if meta.sequenceNr < 10 =>
-            ProcessingSuccess
-          case _ => StorageFailure()
-        }
-      } else {
-        ProcessingSuccess
+}
+//#test
+
+//#set-event-storage-policy
+import akka.persistence.testkit._
+
+class SampleEventStoragePolicy extends EventStorage.JournalPolicies.PolicyType {
```

Comment:
I didn't see any example demoing the policies. Is that something to add to the `persistentTestKit`. Is `withPolicy()` a method on `PersistenceTestKit`? I guess so, but not so sure that the documentation is making that clear for the users. Maybe something to improve on another PR.

Reply:
@manonthegithub is this something you could follow up on?

Reply:
@renatocaval @patriknw it is briefly described in the docs (that you can implement own policy and set it with `withPolicy` method), but there is no example. It seemed straightforward to me. Should we add an example too?

Reply:
If yes, should it be separate branch + PR, or you think of it as part of this PR? If separate, then I would better wait for this one to be merged.

Reply:
@renatocaval @patriknw I meant that there are no examples of how to set policies, but there are of course how to implement.

Reply:
no prob

Reply:
did you create an issue?

Reply:
added #28869